sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -68,7 +68,9 @@ object JDBCRDD extends Logging {
         statement.close()
       }
     } finally {
-      conn.close()
+      if (null != conn) {
+        conn.close()
+      }
```

Comment:
`createConnectionFactory` is called in multiple place. Can you throw an exception when `conn` is `null`?